libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java

```diff
@@ -877,4 +889,30 @@ private Intent getServiceIntent() {
   private boolean isServiceAvailable() {
     return navigationService != null && isBound;
   }
+
+  @Nullable
+  private DirectionsRoute retrieveOfflineRouteFor(@NonNull String directionsUri) {
```

Comment:
Maybe we can extract this offline JSON work into a separate class that we can test in isolation?